tests/test_tools.py

```diff
@@ -2174,5 +2175,69 @@ def test_get_tags_invalid():
         get_tags(coll)
         assert e_info == 'ERROR: valid tags are comma or space separated strings of tag names'
 
+@pytest.mark.parametrize(
+    "inputs, expected", [
+        # everything is defined, return 1
+        ([{"repos": {"talk_repo": {"params": {"namespace_id": "35","initialize_with_readme": "false", "name": "repo name"},
+                         "url": "https://gitlab.thebillingegroup.com/api/v4/projects/"},"prum_repo": {}}}], [1]),
+        # repos is not defined, return 0
+        ([{"repos": {}}], [0]),
+        # talk_repo is not defined, return 0
+        ([{"repos": {"talk_repo": {}, "prum_repo": {}}}], [0]),
+        # parameters are not defined
+        ([{"repos": {"talk_repo": {"params": {}, "url": "https://gitlab.thebillingegroup.com/api/v4/projects/"},"prum_repo": {}}}], [0]),
+        # url is not defined
```

Comment:
Another reason to make code readable and get rid of comments is when they are incorrect, which adds confusion doesn't remove it.  Actually, the url *is* defined, it is defined as an empty string.  What do you want your function to check?  One thing it could check is if the 'url' field is missing altogether. Another thing, which is stronger, is that it is not a properly defined URL (has incorrect characters in it for example).  Another thing it can check is if the URL actually exists.  Which is your goal?  The last is the most useful (and the hardest to test....would have to use Mock).